sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java

```diff
@@ -488,14 +488,13 @@ public NewRepositoryImpl setName(@Nullable String s) {
       return this;
     }
 
+    /**
+     * The specified rule key must be unique among the repository. Since version 5.5 it raises
+     * a {@link IllegalArgumentException}, whereas it logs a warning in previous versions.
+     */
```

Comment:
why not use standard javadoc `@throws`
in addition, shouldn't this be documented on interface method `NewExtendedRepository.createRule(String ruleKey)`?